collect_app/src/main/java/org/odk/collect/android/preferences/FormMetadataFragment.java

```diff
@@ -22,13 +23,29 @@
 import static org.odk.collect.android.preferences.PreferenceKeys.KEY_METADATA_EMAIL;
 import static org.odk.collect.android.preferences.PreferenceKeys.KEY_METADATA_PHONENUMBER;
 import static org.odk.collect.android.preferences.PreferenceKeys.KEY_METADATA_USERNAME;
+import static org.odk.collect.android.utilities.PermissionUtils.requestReadPhoneStatePermission;
 
 public class FormMetadataFragment extends BasePreferenceFragment {
     @Override
     public void onCreate(Bundle savedInstanceState) {
 
         super.onCreate(savedInstanceState);
         addPreferencesFromResource(R.xml.form_metadata_preferences);
+
+        requestReadPhoneStatePermission(getActivity(), new PermissionListener() {
+            @Override
+            public void granted() {
+                initPrefs();
+            }
+
+            @Override
+            public void denied() {
+            }
+        }, true);
+        initPrefs();
```

Comment:
Could you just have this unchecked call on `initPrefs()` since it will call `getSingularProperty` which will in turn result in the permissions check? I'm not clear on the purpose of the `requestReadPhoneStatePermission` call above.

Is it perhaps a way to try to avoid having to listen for the event in this fragment? What happens if the user denies the permissions? I presume the values will be blank?

Reply:
First of all, I implemented a small improvement here because I think that if the permission is not granted at least not dangerous properties should be displayed (before everything was blank).

> Could you just have this unchecked call on initPrefs() since it will call getSingularProperty which will in turn result in the permissions check? I'm not clear on the purpose of the requestReadPhoneStatePermission call above

Here it works in a different way in comparison with `FormEntryActivity`. Here I know that I need dangerous properties so I request the permission before. In `FormEntryActivity` I don't know whether a form contains dangerous properties or not so I use RxEventBus.

>I tried this and the behavior is as I expected it to be.

you removed the entire request and the behavior is ok for you? hmmm I don't understand.

Reply:
> at least not dangerous properties should be displayed

Thanks, that's a good improvement.

> In this case I know that I need dangerous properties so I request the permission before.

Fair enough.

> you removed the entire request and the behavior is ok for you?

No, I mean I tried it as you have it and it does seem to work! It just checks it earlier than I expect but I see why now.